### PR TITLE
Features/svyatoslav

### PR DESCRIPTION
**Pull request by SvyatoslavFedynyak**

### CODE REVIEW

File: Task2/Project/Shapes/Commands/Vertex/AddVertex.cs [hunk 1]
```diff
@@ -2,11 +2,15 @@
 {
     public class AddVertex : Interfaces.ICommand
     {
+        /// <summary>
+        /// Property that enable to interract with command name
+        /// <returns>Command name</returns>
```
Comment (iamprovidence) [inline Task2/Project/Shapes/Commands/Vertex/AddVertex.cs]: **Return** tag should not be inside **Summary** tag

File: Task2/Project/Shapes/Models/Canvas.cs [hunk 2]
```diff
@@ -2,106 +2,230 @@
...
+        public Canvas()
+        {
+            shapes = new List<ShapeBase>();
+        }
+
         // PROPERTIES
+        /// <summary>
+        /// Property that enable to get count of shapes
+        /// </summary>
+        /// /// <returns>Count of shapes</returns>
```
Comment (iamprovidence) [inline Task2/Project/Shapes/Models/Canvas.cs]: **///** before return should be removed

File: Task2/Project/Shapes/Models/Canvas.cs [hunk 3]
```diff
@@ -2,106 +2,230 @@
...
         // METHODS
+        /// <summary>
+        /// Method that allow to add new shape to collection
+        /// </summary>
+        /// <param name="shape">Shape that should be added</param>
         public void Add(ShapeBase shape)
         {
-            throw new NotImplementedException();
+            shapes.Add(shape);
+            shapes[Count - 1].PropertyChanged += Canvas_PropertyChanged;
```
Comment (iamprovidence) [inline Task2/Project/Shapes/Models/Canvas.cs]: **shapes[Count - 1]** hard to read. just use **shape** that you have in argument

File: Task2/Project/Shapes/Models/Canvas.cs [hunk 4]
```diff
@@ -2,106 +2,230 @@
...
+        /// <returns>If shape was deleted</returns>
         public bool Remove(ShapeBase shape)
         {
-            throw new NotImplementedException();
+            for (int i = 0; i < shapes.Count; i++)
+            {
+                if (shapes[i] == shape)
+                {
+                    shapes[i].PropertyChanged -= Canvas_PropertyChanged;
+                    shapes.Remove(shape);
```
Comment (iamprovidence) [inline Task2/Project/Shapes/Models/Canvas.cs]: still, why would you remove in cycle? and why do you need Cycle anyway?

File: Task2/Project/Shapes/Models/Canvas.cs [hunk 5]
```diff
@@ -2,106 +2,230 @@
...
-            throw new NotImplementedException();
+            shapes[index].PropertyChanged -= Canvas_PropertyChanged;
+            shapes.RemoveAt(index);
+            OnCollectionChanged(new NotifyCollectionChangedEventArgs(NotifyCollectionChangedAction.Remove));
         }
+        /// <summary>
+        /// Method that removes all shapes with preset predicate
+        /// </summary>
+        /// <param name="match">Predicate with whitch shapes should be removed</param>
+        /// <returns></returns>
```
Comment (iamprovidence) [inline Task2/Project/Shapes/Models/Canvas.cs]: **returns** is empty

File: Task2/Project/Shapes/Models/Canvas.cs (hunk 3 above)
Comment (SvyatoslavFedynyak) [reply Task2/Project/Shapes/Models/Canvas.cs]: but its different objects...

Comment (iamprovidence) [reply Task2/Project/Shapes/Models/Canvas.cs]: no

**Pull request closed without merging**